tests/src/test/scala/matryoshka/helpers/package.scala

```diff
@@ -27,6 +28,21 @@ import scalaz._, Scalaz._
 import scalaz.scalacheck.ScalaCheckBinding.{GenMonad => _, _}
 
 package object helpers extends SpecificationLike with Discipline {
+  implicit def cogenFree[F[_]: Traverse, A: Monoid : Cogen]: Cogen[Free[F, A]] =
+    Cogen[A] contramap (_.suml)
+
+  implicit def cogenFunctor: Contravariant[Cogen] = new Contravariant[Cogen] {
```

Comment:
There is `Contravariant[Cogen]`
https://github.com/scalaz/scalaz/blob/v7.2.7/scalacheck-binding_1_13/src/main/scala/scalaz/scalacheck/ScalaCheckBinding1.scala